src/Core/src/Platform/Android/WrapperView.cs

```diff
@@ -83,12 +84,41 @@ protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
 
 		public override void RequestLayout()
 		{
-			// Redraw shadow (if exists)
-			_invalidateShadow = true;
+			// Redraw shadow (if exists and required)
+			_invalidateShadow = ShouldInvalidateShadow();
 
 			base.RequestLayout();
 		}
 
+		bool ShouldInvalidateShadow()
+		{
+			if (_shadowBitmap == null)
+				return true;
+
+			if (_viewBounds == null)
+				return false;
+	
+			var viewHeight = _viewBounds.Height();
+			var viewWidth = _viewBounds.Width();
+
+			if (GetChildAt(0) is AView child)
+			{
+				if (viewHeight == 0)
+					viewHeight = child.MeasuredHeight;
+
+				if (viewWidth == 0)
+					viewWidth = child.MeasuredWidth;
+			}
+
+			viewHeight += MaximumRadius;
+			viewWidth += MaximumRadius;
+
+			if (viewHeight == _lastShadowSize.Height && viewWidth == _lastShadowSize.Width)
+				return false;
+
+			return true;
+		}
```

Comment:
Should you consider writing this method entirely in Java? `_viewBounds` is an `Android.Graphics.Rect`.

So then each of these is a C# to Java transition:
* `_viewBounds.Height()`
* `_viewBounds.Width()`
* `GetChildAt(0)`
* `child.MeasuredHeight`
* `child.MeasuredWidth`

So if the whole method was Java, it could be 1 transition instead?